### PR TITLE
feat: karma circle refactor ping-3529 ping-3530 ping-3525

### PR DESCRIPTION
**Pull request by yusufputra**
### Summary by CodeRabbit

- New Feature: Updated the Karma score visualization with a new color scheme to enhance user understanding and engagement.
- Refactor: Enhanced the `KarmaScore` component's text content for better clarity on what the Karma score represents.
- Style: Adjusted the font weight, padding, and color of text elements in the `KarmaScore` component for improved readability.
- Refactor: Replaced the `CircleGradient` component with the `ProfilePicture` component in various screens for consistent profile picture rendering.
- New Feature: Added a `withKarma` prop to the `DomainList` and `Header` components, enabling conditional rendering of the karma score.

### CODE REVIEW

**Comment by bastian-purrer** [comment]
Image description   CodeRabbit



Currently reviewing new changes in this PR...

<details>
<summary>Commits</summary>
Files that changed from the base of the PR and between 5a2e71a9cdfb3ff23b4d8bc1a1b5df668e0b2e1a and cd78a1370d5d252bf0ccdc394a27f6221e78fd9f commits.
</details>

<details>
<summary>Files selected (7)</summary>

* src/components/Karma/utils.ts (1)
* src/screens/DiscoveryScreenV2/elements/DiscoveryItemList.js (3)
* src/screens/DiscoveryScreenV2/fragment/UsersFragment.js (1)
* src/screens/OtherProfile/index.js (2)
* src/screens/ProfileScreen/elements/KarmaScore.tsx (2)
* src/screens/ProfileScreen/elements/ProfilePicture.js (2)
* src/screens/ProfileScreen/index.js (1)
</details>


<details>
<summary>Files ignored due to filter (1)</summary>

* ios/Podfile.lock

</details>





---

File: src/screens/ProfileScreen/elements/KarmaScore.tsx [hunk 1]
```diff
@@ -57,7 +57,7 @@ export const KarmaScore = ({score}: KarmaScoreProps) => {
                   lineHeight: normalize(24),
                   marginBottom: normalize(8)
                 }}>
-                Your Karma Score
+                What is my Karma Score?
```
Comment (bastian-purrer) [inline src/screens/ProfileScreen/elements/KarmaScore.tsx:60]: Image description   CodeRabbit

The change in the text from "Your Karma Score" to "What is my Karma Score?" seems more informative and user-friendly. It gives a clear indication that the following text will explain what the Karma score is.
```diff
- Your Karma Score
+ What is my Karma Score?
```

File: src/screens/ProfileScreen/elements/KarmaScore.tsx [hunk 2]
```diff
@@ -69,12 +69,13 @@ export const KarmaScore = ({score}: KarmaScoreProps) => {
                 }}>
                 The higher your Karma score, the higher your visibility on the platform. Get
                 rewarded for positive contributions to the community - and avoid being blocked by
-                others!
+                others! Your Karma score will be shown as a circle around your profile picture,
+                including for your anonymous posts.
```
Comment (bastian-purrer) [inline src/screens/ProfileScreen/elements/KarmaScore.tsx:73]: Image description   CodeRabbit

The addition of information about where the Karma score will be displayed is useful for users. It helps them understand where they can find their Karma score on the platform.
```diff
- others!
+ others! Your Karma score will be shown as a circle around your profile picture,
+ including for your anonymous posts.
```

File: src/screens/ProfileScreen/elements/ProfilePicture.js [hunk 3]
```diff
@@ -24,36 +27,55 @@ const ProfilePicture = ({
...
+              style={styles.profileImage(size, width)}
               source={{
                 uri: profilePicPath ? `${profilePicPath}` : DEFAULT_PROFILE_PIC_PATH
               }}
               resizeMode={FastImage.resizeMode.stretch}
             />
             {renderAddIcon()}
           </View>
-        </CircleGradient>
+        )}
```
Comment (bastian-purrer) [inline src/screens/ProfileScreen/elements/ProfilePicture.js:60]: Image description   CodeRabbit

The code for rendering the `CircleGradient` and the `View` components is almost identical, except for the wrapping `CircleGradient` component when `withKarma` is true. This repetition can be avoided by using a single render function that conditionally wraps the content with `CircleGradient` based on the `withKarma` prop. Here's how you could refactor this:

```diff
- {withKarma ? (
-   <CircleGradient
-     fill={karmaScore}
-     size={normalize(size)}
-     width={normalize(width)}
-     testId="images">
-     <View style={styles.profileImageContainer(size)}>
-       <Image
-         testId="images"
-         style={styles.profileImage(size, width)}
-         source={{
-           uri: profilePicPath ? `${profilePicPath}` : DEFAULT_PROFILE_PIC_PATH
-         }}
-         resizeMode={FastImage.resizeMode.stretch}
-       />
-       {renderAddIcon()}
-     </View>
-   </CircleGradient>
- ) : (
-   <View style={styles.profileImageContainer(size)} testId="images">
-      <Image
-       testId="images"
-       style={styles.profileImage(size, width)}
-        source={{
-          uri: profilePicPath ? `${profilePicPath}` : DEFAULT_PROFILE_PIC_PATH
-        }}
-        resizeMode={FastImage.resizeMode.stretch}
-      />
-      {renderAddIcon()}
-    </View>
- )}
+ {renderProfileContent(withKarma)}
```

And then define the `renderProfileContent` function as follows:

```javascript
const renderProfileContent = (withKarma) => {
  const content = (
    <View style={styles.profileImageContainer(size)} testId="images">
      <Image
        testId="images"
        style={styles.profileImage(size, width)}
        source={{
          uri: profilePicPath ? `${profilePicPath}` : DEFAULT_PROFILE_PIC_PATH
        }}
        resizeMode={FastImage.resizeMode.stretch}
      />
      {renderAddIcon()}
    </View>
  );

  return withKarma ? (
    <CircleGradient
      fill={karmaScore}
      size={normalize(size)}
      width={normalize(width)}
      testId="images">
      {content}
    </CircleGradient>
  ) : content;
};
```

This way, you avoid duplicating the code for rendering the image and the add icon.

File: src/screens/DiscoveryScreenV2/elements/DiscoveryItemList.js [hunk 4]
```diff
@@ -10,8 +10,8 @@ import {
 } from 'react-native';
 import MemoDomainProfilePicture from '../../../assets/icon/DomainProfilePictureEmptyState';
 import {COLORS} from '../../../utils/theme';
-import {CircleGradient} from '../../../components/Karma/CircleGradient';
-import {fonts, normalize} from '../../../utils/fonts';
+import {fonts} from '../../../utils/fonts';
+import ProfilePicture from '../../ProfileScreen/elements/ProfilePicture';
```
Comment (bastian-purrer) [inline src/screens/DiscoveryScreenV2/elements/DiscoveryItemList.js:14]: Image description   CodeRabbit

The import of `ProfilePicture` component is a good replacement for the `CircleGradient` component. This change aligns with the PR's goal of modifying Karma score representation.

```diff
- import {CircleGradient} from '../../../components/Karma/CircleGradient';
- import {fonts, normalize} from '../../../utils/fonts';
+ import {fonts} from '../../../utils/fonts';
+ import ProfilePicture from '../../ProfileScreen/elements/ProfilePicture';
```

File: src/screens/DiscoveryScreenV2/elements/DiscoveryItemList.js [hunk 5]
```diff
@@ -31,7 +31,8 @@ const DomainList = (props) => {
     onPressBody,
     DefaultImage,
     isCommunity,
-    isBlockedSection
+    isBlockedSection,
+    withKarma
```
Comment (bastian-purrer) [inline src/screens/DiscoveryScreenV2/elements/DiscoveryItemList.js:35]: Image description   CodeRabbit

Adding the `withKarma` prop to the destructuring of props is necessary for the conditional rendering of the Karma score in the `ProfilePicture` component. Good addition.

```diff
-    isBlockedSection
+    isBlockedSection,
+    withKarma
```

File: src/screens/DiscoveryScreenV2/elements/DiscoveryItemList.js [hunk 6]
```diff
@@ -80,34 +81,13 @@ const DomainList = (props) => {
...
-        }}
-        style={styles.profilepicture}
-        width={48}
-        height={48}
+      <ProfilePicture
+        profilePicPath={item.image}
+        karmaScore={item.karmaScore}
+        size={51}
+        width={3}
+        withKarma={withKarma}
```
Comment (bastian-purrer) [inline src/screens/DiscoveryScreenV2/elements/DiscoveryItemList.js:90]: Image description   CodeRabbit

The new implementation of `renderProfilePicture` function is more concise and maintainable. It uses the `ProfilePicture` component and passes necessary props including `withKarma` for conditional rendering of the Karma score. 

```diff
-    if (item.karmaScore) {
-      return (
-        <CircleGradient
-          testId="images"
-          fill={item.karmaScore}
-          size={normalize(51)}
-          width={normalize(3)}>
-          <Image
-            testId="images"
-            source={{
-              uri: item.image
-            }}
-            style={styles.profilepicture}
-            width={48}
-            height={48}
-          />
-        </CircleGradient>
-      );
-    }
-     return (
-      <Image
-        testId="images"
-        source={{
-          uri: item.image
-        }}
-        style={styles.profilepicture}
-        width={48}
-        height={48}
-       />
-     );
+    return (
+      <ProfilePicture
+        profilePicPath={item.image}
+        karmaScore={item.karmaScore}
+        size={51}
+        width={3}
+        withKarma={withKarma}
+      />
+    );
```

**Review by bastian-purrer — COMMENTED** [review]
Image description   CodeRabbit

<details>
<summary>Commits</summary>
Files that changed from the base of the PR and between 5a2e71a9cdfb3ff23b4d8bc1a1b5df668e0b2e1a and cd78a1370d5d252bf0ccdc394a27f6221e78fd9f commits.
</details>

<details>
<summary>Files selected (7)</summary>

* src/components/Karma/utils.ts (1)
* src/screens/DiscoveryScreenV2/elements/DiscoveryItemList.js (3)
* src/screens/DiscoveryScreenV2/fragment/UsersFragment.js (1)
* src/screens/OtherProfile/index.js (2)
* src/screens/ProfileScreen/elements/KarmaScore.tsx (2)
* src/screens/ProfileScreen/elements/ProfilePicture.js (2)
* src/screens/ProfileScreen/index.js (1)
</details>


<details>
<summary>Files ignored due to filter (1)</summary>

* ios/Podfile.lock

</details>






<details>
<summary>Files skipped from review due to trivial changes (3)</summary>

* src/components/Karma/utils.ts
* src/screens/OtherProfile/index.js
* src/screens/ProfileScreen/index.js

</details>

<details>
<summary>Review comments generated (9)</summary>

* Review: 6
* LGTM: 3

</details>

---

<details>
<summary>Tips</summary>

### Chat with Image description  CodeRabbit Bot (`@coderabbitai`)
- Reply on review comments left by this bot to ask follow-up questions. A review comment is a comment on a diff or a file.
- Invite the bot into a review comment chain by tagging `@coderabbitai` in a reply.

### Code suggestions
- The bot may make code suggestions, but please review them carefully before committing since the line number ranges may be misaligned. 
- You can edit the comment made by the bot and manually tweak the suggestion if it is slightly off.

### Pausing incremental reviews
- Add `@coderabbitai: ignore` anywhere in the PR description to pause further reviews from the bot.

</details>

**Review by asaadam — APPROVED** [review]

**Pull request closed without merging**